### PR TITLE
[DEVTOOLS-91] Adding connection limiter

### PR DESCRIPTION
**Pull request by mmerrell**
Adding entries to .gitignore to aid in Mac development

[DEVTOOLS-91] Adding connection limiter
[DEVTOOLS-91] Updating README and config-example.json

### CODE REVIEW

**Comment by mmerrell** [comment]
This PR should address issue-115

**Comment by kellegous** [comment]
Also #113. Thanks so much for fixing this, I skimmed the change and it looks good. Token bucket in a channel was exactly what I planned to do. I'm a little behind in integrating PR's but I hope to get caught up very soon.

File: searcher/searcher.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func init() {
 // NOTE: The keys in the searcher map will be normalized to lower case, but not such
 // transformation will be done on the error map to make it easier to match those errors
 // back to the original repo name.
-func MakeAll(cfg *config.Config) (map[string]*Searcher, map[string]error, error) {
+func MakeAll(cfg *config.Config, connectionLimiter) (map[string]*Searcher, map[string]error, error) {
```
Comment (kellegous) [inline searcher/searcher.go]: Finally got around to merge this tonight, but I'm finding it doesn't build.

**Comment by kellegous** [comment]
@mmerrell  Can you take a look at #118 and make sure it still solves the problem you are having. I changed a couple of things:

 1. The concurrency limit is about more than just the outgoing connections. We need a limiter on indexers in general. So I have renamed the config to `max-concurrent-indexers`.
 2. I move the creation of the token bucket out of main and put it into the searcher. I also created a type to wrap the channel so we can have `Acquire` & `Release` semantics. I also moved a good chunk of the reindexing loop into its own function to make this a bit cleaner.

**Pull request merged**